content/articles/pull-requests.md

```diff
@@ -0,0 +1,91 @@
+   - Set the state as `Submitted`
+
+#### Then the reviewer
+
+1. Check that you are an appropriate reviewer
+1. Look for:
+   * does the code meet a need or requirement (should be clear between the ticket and the PR paragraph)
+   * formatting, line lengths and lint issues
+   * is the code generally understandable? If not escalate via the PR, slack and finally in person if necessary
+   * does the code look like it would work?
+   * does it pass on CI?
+   * does it work when run locally? (not possible for all components)
+   * how are the tests?  is all new code tested?
+   * are there obvious inefficiencies (if appropriate)
+   * for web apps, check that they render properly
+   * is it documented, and are odd bits of code well commented if they can't be simplified
+
+### The process
+
+* Not too many rounds of review. One round of requested changes is expected, two is going to happen sometimes, but more than that suggests a process we could have improved
+* If there are conflicts (most likely, due to another PR being merged first) then it is the **author** wno needs to resolve this.  Merge `master` into the PR (follow instructions on the PR itself) and then check that things still work as expected
+* After acceptance then the **reviewer**
+  1. approves the PR
+  1. merges the PR
+  1. updates ticket to unassign themselves and move state to "Completed" or "Ready to deploy" as appropriate
+  1. check that `master` still works on CI
+
+
+### Expectations
+
+What the different parties expect to get out of the PR review process
+
+#### Author
+
+All authors appreciate feedback on:
+
+* What have I done wrong, or what could I do better?
+* What have I overlooked? Are there corner cases I have missed?
+* What tests are missing?
+* Is my code readable and idiomatic?
+* For UI code, how is the look and feel?
+
+Expected time to initial response is 1/2 a day to 1 day (the review might not be completed in this time, but this should be indicated).  Once the review process is started, re-review is expected to be quicker as it should address ever-smaller issues.
+
+#### Reviewer
+
+* Explanations and descriptions should be provided for nontrivial sections
+* Smaller is better.  400 lines is a ballpark figure
+* CI should be passing before the review is assigned
+* Each PR should be conceptually distinct
+
+#### General
+
+* YouTrack is the source of truth for the state of a review (i.e., what needs doing next)
+* Both parties are responsible for checking CI throughout the process
+* One of the main aims of the review process is to grow and maintain the team understanding of our codebases
+* All code should be reviewed eventually
+* The process is expected to be constructive.  Reviewers must always be polite, and authors must take suggested improvements in good faith
+* Many suggestions will not be incorporated into the PR in question but may become new tickets
+
+#### Outstanding issues
+
+* We would benefit from a style guide, probably one per language
```

Comment:
Should we also have a PR verbiage guide? I think we all do a great job of communicating at the moment but if this protocol is intended as a long-lived resource that will outlast current team makeup, be used in onboarding etc?

Reply:
What do you mean by "PR verbiage guide"?

Reply:
I mean like a code of conduct type doc - phrasing and language guidelines for online discussions. I remember you sent me someone else's example of this ages ago but I can't remember who's it was.

Reply:
cool, added a note which mostly just links to https://vimc.myjetbrains.com/youtrack/issue/RESIDE-11